libgearman-server/plugins/queue/redis/queue.cc

```diff
@@ -202,7 +202,7 @@ static gearmand_error_t _hiredis_add(gearman_server_st *, void *context,
   build_key(key, unique, unique_size, function_name, function_name_size);
   gearmand_log_debug(GEARMAN_DEFAULT_LOG_PARAM, "hires key: %u", (uint32_t)key.size());
 
-  redisReply *reply= (redisReply*)redisCommand(queue->redis(), "SET %b %b", &key[0], key.size(), data, data_size);
+  redisReply *reply= (redisReply*)redisCommand(queue->redis(), "SET %s %b", &key[0], data, data_size);
```

Comment:
I had to change that because [this](https://github.com/flavio/gearmand/blob/e99429ef7b39f02ed891f45e344d212db40d6ebb/libgearman-server/plugins/queue/redis/queue.cc#L287) `GET` request would have always returned an empty result. The key searched would not have been null terminated. We could change that to `GET %b`...

Also inside of Redis the key would be stored as `_gear_-foo-123\00\00\00\00` which is not so nice :)

We are in full control over the key generation, hence there should not be relatively safe